### PR TITLE
KOA-4927 Add compose BpkRadioButton

### PR DESCRIPTION
**Pull request by marianeum**
Added a new `BpkRadioButton` component matching the current view version of this

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] Component `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: app/src/main/java/net/skyscanner/backpack/demo/compose/BackpackPreview.kt [hunk 1]
```diff
@@ -39,8 +38,8 @@ fun BackpackPreview(
   BackpackDemoTheme {
     CompositionLocalProvider(*providers) {
       Surface(
-        modifier = modifier
-          .background(background.takeOrElse { BpkTheme.colors.background }),
+        color = background.takeOrElse { BpkTheme.colors.background },
```
Comment (marianeum) [inline app/src/main/java/net/skyscanner/backpack/demo/compose/BackpackPreview.kt:41]: we were setting the color via the modifier which meant it wasn't actually picked up correctly - set color directly

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/radiobutton/BpkRadioButton.kt [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun BpkRadioButton(
+  selected: Boolean,
+  onClick: (() -> Unit)?,
+  modifier: Modifier = Modifier,
+  enabled: Boolean = true,
+  interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+) {
+  CompositionLocalProvider(LocalMinimumTouchTargetEnforcement provides false) {
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/radiobutton/BpkRadioButton.kt:49]: this removes the forced 48dp touch target to match our current component

Comment (bvitaliyg) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/radiobutton/BpkRadioButton.kt:49]: Let's add a comment explaining this decision

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/radiobutton/BpkRadioButton.kt [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+) {
+  val rowModifier = if (onClick != null) {
+    modifier.clickable(interactionSource = interactionSource, indication = null, onClick = onClick)
+  } else {
+    modifier
+  }
+  Row(
+    horizontalArrangement = Arrangement.spacedBy(BpkDimension.Spacing.Sm),
+    modifier = rowModifier,
+    verticalAlignment = Alignment.CenterVertically
```
Comment (bvitaliyg) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/radiobutton/BpkRadioButton.kt:81]: Trailing commas?

File: app/src/main/java/net/skyscanner/backpack/demo/compose/RadioButtonStory.kt [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    DefaultCheckedRadioButtonExample(selected = selectedIndex == 1, onClick = { selectedIndex = 1 })
+
+    DisabledUnCheckedRadioButtonExample()
+    DisabledCheckedRadioButtonExample()
+  }
+}
+
+@Preview
+@Composable
+fun DefaultUncheckedRadioButtonExample(selected: Boolean = false, onClick: (() -> Unit)? = null) {
```
Comment (bvitaliyg) [inline app/src/main/java/net/skyscanner/backpack/demo/compose/RadioButtonStory.kt:54]: I still believe we don't need to create these previews, but it's not a problem for this PR.

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/radiobutton/BpkRadioButton.kt (hunk 3 above)
Comment (marianeum) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/radiobutton/BpkRadioButton.kt:81]: how have you configured your AS? i've got trailing commas enabled, but it's not adding them in most places. looks like ktlint now has an experimental rule for it, so hopefully it'll come soon 🎉

File: app/src/main/java/net/skyscanner/backpack/demo/compose/RadioButtonStory.kt (hunk 4 above)
Comment (frugoman) [reply app/src/main/java/net/skyscanner/backpack/demo/compose/RadioButtonStory.kt:54]: You can basically iterate code without building the entire app, dont see why we wouldn't want them

**Review by frugoman — APPROVED** [review]

**Review by bvitaliyg — APPROVED** [review]

**Pull request merged**